File: GrooveOn.Models/ResponseObjects/ExistingAlbumTrackResponse.cs
namespace GrooveOn.Model.Responses
{
    public class ExistingAlbumTrackResponse
    {
        public string ExternalTrackId { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public bool AlreadyExists { get; set; }
    }
}

File: UI/grooveon_desktop/lib/models/response/existing_album_track_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'existing_album_track_response.g.dart';

@JsonSerializable()
class ExistingAlbumTrackResponse {
  final String externalTrackId;
  final String title;
  final bool alreadyExists;

  ExistingAlbumTrackResponse({
    required this.externalTrackId,
    required this.title,
    required this.alreadyExists,
  });

  factory ExistingAlbumTrackResponse.fromJson(Map<String, dynamic> json) =>
      _$ExistingAlbumTrackResponseFromJson(json);

  Map<String, dynamic> toJson() => _$ExistingAlbumTrackResponseToJson(this);
}